TicketingSystem.Tests/RepositoryTests/CartRepositoryTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using FluentAssertions;
+using Moq;
+using Moq.EntityFrameworkCore;
+using TicketingSystem.ApiService.Repositories.CartRepository;
+using TicketingSystem.Common.Context;
+using TicketingSystem.Common.Model.Database.Entities;
+
+namespace TicketingSystem.Tests.RepositoryTests
+{
+    [TestClass]
+    public class CartRepositoryTests
+    {
+        private readonly Mock<TicketingDbContext> _mockContext;
+        private readonly CartRepository _cartRepository;
+
+        public CartRepositoryTests()
+        {
+            _mockContext = new Mock<TicketingDbContext>();
+            _cartRepository = new CartRepository(_mockContext.Object);
+        }
+
+        [TestMethod]
+        public async Task FirstOrDefaultWithTicketsAsync()
+        {
+            // Arrange
+            Guid cartId = Guid.Empty;
+
+            var ticket = new Ticket
+            {
+                EventId = 1,
+                PriceCategoryId = 1,
+                SeatId = 1,
+                Status = Common.Model.Database.Enums.TicketStatus.Free,
+                CartId = cartId
+            };
+            var tickets = new List<Ticket> { ticket };
+
+            var cart = new Cart
+            {
+                CartId = cartId,
+                PersonId = 1,
+            };
+            var carts = new List<Cart> { cart };
+
+            _mockContext.Setup(x => x.Tickets).ReturnsDbSet(tickets);
```

Comment:
Frankly speaking, I do not understand the idea behind such tests. `DbSet` is backed by `List`. It will work for simple cases that correlate with Linq extensions for `IEnumerable`. But it will not work for more complex scenarios that may use DbFunctions or Expressions that are later translated into some specific SQL.